component/azstorage/utils.go

```diff
@@ -575,7 +575,7 @@ func getMD5(fi *os.File) ([]byte, error) {
 }
 
 func autoDetectAuthMode(opt AzStorageOptions) string {
-	if opt.ApplicationID != "" || opt.ResourceID != "" {
+	if opt.ApplicationID != "" || opt.ResourceID != "" || opt.ObjectID != "" {
```

Comment:
We are saying if nothing is set then also MSI can work. Do we want to use msi as default or we can skip that case here and ask user to explicitly configure the auth mode to be msi in such case (where no config is available and vm is having the identity assigned that can access the storage account).

Reply:
Default to msi if nothing is passed has been committed